### DISCUSSION TITLE
Rotate on spot only

### DISCUSSION THREAD

**Discussion by Gabby604** [google/model-viewer#3584] [General]
Hi,

I don't know how to explain... but I would like that my model can rotate on the X Axis only but at the same spot (same thing when you turn on ourselves, you stay at the same place). For now, there is an offset when the character rotates. The character will be in an environment like a terrain so he can't move from his position.
What function does I have to use for that ?

Thank you !

**Comment by elalish** [comment]
I'm having some trouble following your question, but I believe you want some combination of https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-cameraTarget (the point it turns around), https://modelviewer.dev/docs/#entrydocs-stagingandcameras-methods-resetTurntableRotation and maybe https://modelviewer.dev/docs/#entrydocs-scenegraph-attributes-orientation

**Reply by Gabby604** [reply]
yeah it's difficult to explain ^^ But in fact, it's because of my model. I've download them from Sketchab (it's a pack) and some models rotate correctly at the same place, and other are offset. I don't know why.. I tried to edit them in Blender but still the same issue..
In my example, the Dog rotates on himself, but the cow is offset when I rotate her.

![2022-07-18_12-21-21](https://user-images.githubusercontent.com/107591631/179600714-8d2b537b-d97c-418b-b95f-be5c665136c1.jpg)

**Reply by elalish** [reply]
Are you even using `<model-viewer>` for this? Are they multiple models or one?

**Reply by Gabby604** [reply]
Yes I'm using `<model-viewer>` and there are 4 characters. Some of them work and not others..

**Reply by elalish** [reply]
I mean, `<model-viewer>` has no API to rotate one mesh within a GLB, so I'm really not sure what you're doing. Maybe you can link in some code or an example?

**Reply by Gabby604** [reply]
Can I send you a link in private?

**Reply by elalish** [reply]
No, but you can explain or link a simpler repro.

**Reply by Gabby604** [reply]
Ok anyway it's just a part of the whole website. So you can try here to see what happens : https://shivaconception.com/3d/
But as I said, I think it's because of the models. Some are okay but not others and I can't find a solution for now..

**Reply by elalish** [reply]
Ah, thank you for the demo, that's cute! Okay, I understand now; each animal is a separate MV element and they're all on a big continuous background image. That's a clever idea! I'd bet the issue with the cow is a transform on your animation - we calculate the bounds based on the resting pos, so if your animation translates it, then it'll be off like you see. If you can't manage to fix the GLB itself, you can adjust https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-cameraTarget instead to override our default centering.

**Reply by Gabby604** [reply]
Ok thank you, I will try that. Because when I import the Cow in Blender, everything is okay I think. The only solution I found is to import the GLTF from Sketchfab into Unity and export it in a new FBX, then import it into Blender and export it again in GLTF. Now the Cow rotates on herself BUT... she's smaller for a reason I ignore.. I can't scale her.. Very weird.

Well, thanks again for you help ! :) I will try to adjust the CameraTarget to see what happens.
